File: testplugin/src/main/java/com/zj/testplugin/MainActivity.java
package com.zj.testplugin;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.e("zhang", "插件的MainActivity启动");

//        setContentView(R.layout.activity_main);
    }
}

File: app/src/main/java/com/zj/plugindemo/MainActivity.kt
package com.zj.plugindemo

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

    }

    fun openPlugin(view: View) {
        try {
            val clazz = Class.forName("com.zj.testplugin.Test")
            val print = clazz.getMethod("test")
            print.invoke(null)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}